src/query/query_hash_scan.c

```diff
@@ -1924,7 +1924,12 @@ fhs_binary_search_bucket (THREAD_ENTRY * thread_p, PAGE_PTR bucket_page_p, PGSLO
 	}
       else
 	{
-	  low = (flag > 0) ? middle + flag : middle + 1;
+	  if (flag > 0)
+	    {
+	      /* go to last slot using flag */
+	      middle = middle + flag - 1;
+	    }
+	  low = middle + 1;
```

Comment:
```suggestion
	  middle += (flag > 0) ? flag - 1 : 0; /* go to last slot using flag */
	  low = middle + 1;
```
the same code format as before